data/core/macros/unit-utils.cfg

```diff
@@ -253,3 +253,766 @@
         transform_to={TYPE}
     [/transform_unit]
 #enddef
+
+#define FORMATION ID GREATER_THAN COUNT
+    [chance_to_hit]
+        id={ID}
+        apply_to=opponent
+        sub=10
+        [filter_base_value]
+            greater_than_equal_to={GREATER_THAN}
+        [/filter_base_value]
+        [filter_self]
+            [filter_adjacent]
+                is_enemy=no
+                count={COUNT}
+                ability=formation
+            [/filter_adjacent]
+        [/filter_self]
+    [/chance_to_hit]
+#enddef
+
+# Support ability
+#define ABILITY_SUPPORT_EVENTS
```

Comment:
Both ABILITY_SUPPORT_EVENTS and FORMATION sound too generic for names of macros in core - too likely that UMC will define a similarly-named event.

Reply:
Technically that's not a problem for the ability macros. The core files are read first, the core units will be correctly created. When the add-on is read, it will define a macro with the same name, which overrides this one (in the scope of that add-on only, not interfering with other add-ons). The add-on's units will be created with the add-on's macro, so also correctly.
And when overriding existing macros, a warning is given in stderr.

However, one problem will be the event macros, there the add-on needs the core version of the macro too. Currently it's set up that these macros must be included into every scenario (or era, or campaign) manually, which breaks the unspoken principle that one can just use units in a UMC scenario without having to worry about further things. That's an issue by itself – maybe we find a solution for this.

Reply:
The macros should have a reasonable name though, that is, a somewhat generic one, because they can be used on any unit now, not only UTBS ones.